Cython/Compiler/FusedNode.py

```diff
@@ -613,13 +633,16 @@ def make_fused_cpdef(self, orig_py_func, env, is_def):
                     int __Pyx_Is_Little_Endian()
             """)
         decl_code.indent()
-
         pyx_code.put_chunk(
             u"""
                 def __pyx_fused_cpdef(signatures, args, kwargs, defaults):
                     # FIXME: use a typed signature - currently fails badly because
                     #        default arguments inherit the types we specify here!
 
+                    cdef list dest_sig, sigindex_matches, sigindex_candidates, found_matches, found_candidates, search_list, candidates
```

Comment:
You don't need to declare all of these. Cython has assignment based type inference, so anything that is only assigned a single type will automatically be typed inside of the function.

Reply:
Oh, cool. Does it also recurse in that names will be still typed if they're sometimes assigned based on other names that are only ever assigned to a single type?

Reply:
Yes it should do. You can usually verify this (if you need to) by either looking at the C code or by using Visitor.PrintTree